src/NServiceBus.Testing/Fakes/TestableIncomingContext.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace NServiceBus.Testing
+{
+    using NServiceBus.ObjectBuilder;
+    using NServiceBus.Pipeline;
+
+    /// <summary>
+    /// Base implementation for contexts implementing <see cref="IIncomingContext"/>.
+    /// </summary>
+    public abstract class TestableIncomingContext : TestableMessageProcessingContext, IIncomingContext
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="TestableIncomingContext"/>.
+        /// </summary>
+        protected TestableIncomingContext(IMessageCreator messageCreator = null) : base(messageCreator)
+        {
+        }
+
+        IBuilder IBehaviorContext.Builder => GetBuilder();
+
+        /// <summary>
+        /// A fake <see cref="IBuilder"/> implementation. If you want to provide your own <see cref="IBuilder"/> implementation override <see cref="GetBuilder"/>.
+        /// </summary>
+        public FakeBuilder Builder { get; set; } = new FakeBuilder();
+
+        /// <summary>
+        /// Selects the builder returned by <see cref="IBehaviorContext.Builder"/>. Override this method to provide your custom <see cref="IBuilder"/> implementation.
+        /// </summary>
+        protected virtual IBuilder GetBuilder()
```

Comment:
not really because of the explicit interface implementation which needs to be on this class. Moving the testable property and the `GetBuilder` away doesn't add value imho.